### PR TITLE
{vis}[foss-2018a] MathGL 2.4.1 (REVIEW)

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

**Comment by boegel** [comment]
@hajgato Given the overlap, I'd like to get #6018 merged first.

**Comment by boegel** [comment]
@hajgato #6018 is merged, please sync this with current `develop`?

**Comment by hajgato** [comment]
@boegel `develop` synced.

File: easybuild/easyconfigs/m/MathGL/MathGL-2.4.1-foss-2018a.eb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+configopts += '-Denable-jpeg=ON '
+configopts += '-Denable-gsl=ON '
+configopts += '-Denable-hdf5=ON '
+configopts += '-Denable-glut=ON '
+configopts += '-Denable-fltk=ON '
+configopts += '-Denable-qt5=ON '
+
+sanity_check_paths = {
+    'files': ['bin/mglconv', 'bin/mgllab', 'bin/mglview', 'bin/udav'] +
+    ['lib/libmgl%s.%s' % (x, y) for x in ['', '-fltk', '-glut', '-qt5', '-qt'] for y in ['a', SHLIB_EXT]],
```
Comment (boegel) [inline easybuild/easyconfigs/m/MathGL/MathGL-2.4.1-foss-2018a.eb]: @hajgato Style nitpicking: please fix indent, align the `[` with the one on the line above (and make sure you stick below 120 chars per line, but should be fine)

File: easybuild/easyconfigs/m/MathGL/MathGL-2.4.1-foss-2018a.eb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+configopts += '-Denable-opengl=ON '
+configopts += '-Denable-openmp=ON '
+configopts += '-Denable-zlib=ON '
+configopts += '-Denable-png=ON '
+configopts += '-Denable-jpeg=ON '
+configopts += '-Denable-gsl=ON '
+configopts += '-Denable-hdf5=ON '
+configopts += '-Denable-glut=ON '
+configopts += '-Denable-fltk=ON '
+configopts += '-Denable-qt5=ON '
```
Comment (boegel) [inline easybuild/easyconfigs/m/MathGL/MathGL-2.4.1-foss-2018a.eb]: I'd prefer collapsing these on to a line or two.

Or, if you insist on having one per line, at least order them alphabetically :)

File: easybuild/easyconfigs/m/MathGL/MathGL-2.4.1_FL.patch [hunk 3]
```diff
@@ -0,0 +1,66 @@
+# include direcotry is all caps
```
Comment (boegel) [inline easybuild/easyconfigs/m/MathGL/MathGL-2.4.1_FL.patch]: clarify that this is the `FLTK` include directory (and fix the typo in `direcotry` while you're at it) ;)

don't forget to update the `SHA256` checksum for the patch if you change it

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2653.swalot.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/c99880ac073bb0b5ec5c7e8002f88269 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2106.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/e2a50a9ce15e377ad72da2e152f871f1 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @hajgato!

**Pull request merged**